exporters/tests/test_deserialization.py

```diff
@@ -9,12 +9,77 @@
     FieldTypeCheckError,
     InnerFieldDeserializationErrors,
     MissingFieldError,
+    _extract_dict_types,
+    _extract_list_type,
+    _extract_optional_type,
     deserialize,
     nested,
     retain_source,
 )
 
 
+@pytest.mark.parametrize(
+    "type_",
+    [list[int], str, set[str], Iterable[str], Container[str], tuple[float, float]],
+)  # Iterable and container should not have __getitem__, tuple should pass?
```

Comment:
I don't know if it's worth it to catch every edge case. This was mainly for working with openshift objects, and I don't know when we'd actually expect a tuple!

Reply:
I would add a comment about not supporting tuples then